src/main/java/com/metamx/common/StreamUtils.java

```diff
@@ -81,4 +85,21 @@ public static void copyWithTimeout(InputStream is, OutputStream os, long timeout
       os.write(buffer, 0, n);
     }
   }
+
+  // Return a message digest of the input stream of copy from input stream to output stream. DOES NOT close inputStream
+  public static MessageDigest md5StreamToFile(InputStream inputStream, File outFile) throws IOException
```

Comment:
to keep in line with existing method names, maybe copyToFileWithDigest might be more consistent?
Maybe allowing to pass in the message digest would make this more flexible and not just limited to md5?